esmvalcore/preprocessor/_derive/amoc.py

```diff
@@ -11,7 +11,12 @@ class DerivedVariable(DerivedVariableBase):
     @staticmethod
     def required(project):
         """Declare the variables needed for derivation."""
-        required = [{'short_name': 'msftmyz', 'mip': 'Omon'}]
+        if project == "CMIP5":
+            required = [{'short_name': 'msftmyz', 'mip': 'Omon'}]
+        elif project == "CMIP6":
+            required = [{'short_name': 'msftmz', 'optional': True},
+                        {'short_name': 'msftyz', 'optional': True}]
+
```

Comment:
This fails (probably with a `NameError`) if the project is neither CMIP5 nor CMIP6. I suggest something like (which would follow the previous behavior:

```py
if project == 'CMIP6':
    ...
else:
    ...
```

Reply:
So have the cmip5 case be the catchall/fallback?

Reply:
Yes! If you don't think that's a good idea then I would at least issue a nice error message in case something other than CMIP5 or CMIP6 is given. But since the current version of the code always uses the CMIP5 version, I think adding this as a fallback sounds good to me.

Reply:
I don't think we use any other project for derivation outside the [CMIP5, CMIP6] interval, do we?

Reply:
Derivation is not limited to CMIP5 and CMIP6, so we should definitely expect other values for `project`.

Reply:
at any rate, an `else` followed by a nice exception raised is a good idea no matter :+1:

Reply:
> but the project given to the derived variable by the custom cmor class is always either CMIP5 or 6 - or am I living in the past?

Very good question! I'm not sure if the `project` passed to this function is determined by the CMOR class or the project of the data (I suspect the latter). However, in theory it could definitely be CMIP3 since we also have a [`CMIP3Info`](https://github.com/ESMValGroup/ESMValCore/blob/b902e580e06690cbc7f8a568ec9d94b7dbd087a1/esmvalcore/cmor/table.py#L772) class!

Reply:
crap, always forget about CMIP3, good point!